src/v/kafka/server/connection_context.cc

```diff
@@ -277,7 +277,7 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
             return ss::make_ready_future<>();
         }
         auto remaining = size - request_header_size
-                         - hdr.client_id_buffer.size();
+                         - hdr.client_id_buffer.size() - hdr.tags_field_size;
```

Comment:
Note to follow up on test coverage here.

Reply:
We can chat about this; usual code review unit test questions; what is covered, what is not covered.

Reply:
This is super crucial so if this is wrong everything blows up. But there is another edit just like this in `src/v/kafka/server/tests/request_parser_test.cc` which is a seperate test binary that reads serialized kafka structures written to a flat file from a python program invoked by CTest. I'm thinking about editing this test to support flex requests, however it depends on `kafka-python` and the latest version [here from Pip](https://pypi.org/project/kafka-python/#history)  2.0.2 doesn't have support for flex as its almost 2 years old now. I have some stashed code that builds a new pex file from the head of whats in github , however i've had to build some of my own python request structures as its not totally complete. Note how here ApiVersionsRequest3 [is missing](https://github.com/dpkp/kafka-python/blob/master/kafka/protocol/admin.py#L59-L64)

Reply:
Yeah I found kafka-python to be lacking when I did some testing with it.

Reply:
I actually found a bug in how it serializes and deserializes tags. I would say whatever is at the tip of dev isn't stable yet